data_structures/stack.py

```diff
@@ -24,16 +24,29 @@ def is_empty(self):
 
 
 class Node:
-    def __init__(self):
-        super().__init__()
-
-
-class StackNode:
     def __init__(self, val):
         self.val = val
         self.next = None
 
 
+class StackNode:
+    def __init__(self):
+        self.head = None
+
+    def push(self, val):
+        new_node = Node(val)
+        new_node.next = self.head
+        self.head = new_node
+        return self.head.val
+
+    def pop(self):
+        self.head = self.head.next
```

Comment:
one minor comment - this will throw an exception if `self.head` is already `None`. Maybe it's better to check / throw your own exception manually.